language/move-binary-format/src/deserializer.rs

```diff
@@ -10,7 +10,9 @@ use std::{collections::HashSet, convert::TryInto, io::Read};
 impl CompiledScript {
     /// Deserializes a &[u8] slice into a `CompiledScript` instance.
     pub fn deserialize(binary: &[u8]) -> BinaryLoaderResult<Self> {
-        CompiledScript::deserialize_no_check_bounds(binary)?.freeze()
+        let script = CompiledScript::deserialize_no_check_bounds(binary)?;
+        BoundsChecker::verify_script(&script)?;
```

Comment:
Probably a good idea to keep this here, even though it is in the other verifier pass. But we should make sure we aren't duplicating work

Reply:
Seeing this kind of makes me want a wrapper around `CompiledModule` that has absolutely zero functionality just something like `UnverifiedModule(CompiledModule)` with `fn into_inner_I_promise_I_know_what_I_am_doing(self) -> CompiledModule { self.0 }`

This is really what CompiledModuleMut should have been if we wanted that sort of type guarantee. Then `verify_module` can strip this thing out and verify it and return the other module... maybe... might not be possible as some spots only verify by reference...